plugins/woocommerce/src/Enums/ProductVisibility.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Enums;
+
+/**
+ * Enum class for all the product visibility.
+ */
+class ProductVisibility {
+	/**
+	 * Visible product.
+	 *
+	 * @var string
+	 */
+	const VISIBLE = 'visible';
+
+	/**
+	 * Product visible in catalog.
+	 */
+	const CATALOG = 'catalog';
+
+	/**
+	 * Product visible in search.
+	 */
+	const SEARCH = 'search';
+
+	/**
+	 * Product invisible.
```

Comment:
Especially if we decide *not* to rename the class, it may be worth flagging what hidden means. Specifically, we should probably highlight that the product is still accessible if navigated to directly.

Reply:
It might be good to use the labels from `wc_get_product_visibility_options` in the doc blocks here, as they succinctly describe what each of the values actually does. But I also agree that adding a bit more to the `hidden` option to clarify that you can still access the product directly would also be good.

Reply:
Makes sense! Improved the docs on https://github.com/woocommerce/woocommerce/pull/54001/commits/41ace7941403a285e9edc0baf71d5636bc0ea734